waku/node/health_monitor.nim

```diff
@@ -0,0 +1,65 @@
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import std/[options, sequtils, strutils, tables], chronos
+
+import waku_node, ../waku_rln_relay
+
+type
+  HealthStatus* = enum
+    INITIALIZING
+    SYNCHRONIZING
+    READY
+    NOT_READY
+    NOT_MOUNTED
+    SHUTTING_DOWN
+
+  HealtReport* = object
+    nodeHealth*: HealthStatus
+    protocolHealth*: Table[string, HealthStatus]
+
+  WakuNodeHealthMonitor* = ref object
+    nodeHealth: HealthStatus
+    node: Option[WakuNode]
+
+proc `$`*(t: HealthStatus): string =
+  result =
+    case t
+    of INITIALIZING: "Initializing"
+    of SYNCHRONIZING: "Synchronizing"
+    of READY: "Ready"
+    of NOT_READY: "Not Ready"
+    of NOT_MOUNTED: "Not Mounted"
+    of SHUTTING_DOWN: "Shutting Down"
+
+const FutIsReadyTimout = 5.seconds
+
+proc getNodeHealthReport*(hm: WakuNodeHealthMonitor): Future[HealtReport] {.async.} =
+  result.nodeHealth = hm.nodeHealth
+
+  if hm.node.isSome(): ## and not hm.node.get().wakuRlnRelay == nil:
+    let getRlnRelayHealth = proc(): Future[HealthStatus] {.async.} =
+      let isReadyStateFut = hm.node.get().wakuRlnRelay.isReady()
```

Comment:
This is supposed to work only if RLN is enabled, right?
For cluster-id's different than 1 this will always be false I think